dubbo-metrics/dubbo-metrics-api/src/main/java/org/apache/dubbo/metrics/model/MetricsSupport.java

```diff
@@ -56,4 +73,120 @@ public static Map<String, String> serviceTags(String appAndServiceName) {
         tags.put(TAG_INTERFACE_KEY, keys[1]);
         return tags;
     }
+
+    public static Map<String, String> methodTags(String names) {
+        String[] keys = names.split("_");
+        if (keys.length != 3) {
+            throw new MetricsNeverHappenException("Error names: " + names);
+        }
+        Map<String, String> tags = applicationTags(keys[0]);
+        tags.put(TAG_INTERFACE_KEY, keys[1]);
+        tags.put(TAG_METHOD_KEY, keys[2]);
+        return tags;
+    }
+
+    public static MetricsKey getMetricsKey(RpcException e) {
+        MetricsKey targetKey;
+        targetKey = MetricsKey.METRIC_REQUESTS_FAILED;
+        if (e.isTimeout()) {
+            targetKey = MetricsKey.METRIC_REQUESTS_TIMEOUT;
+        }
+        if (e.isLimitExceed()) {
+            targetKey = MetricsKey.METRIC_REQUESTS_LIMIT;
+        }
+        if (e.isBiz()) {
+            targetKey = MetricsKey.METRIC_REQUEST_BUSINESS_FAILED;
+        }
+        if (e.isSerialization()) {
+            targetKey = MetricsKey.METRIC_REQUESTS_CODEC_FAILED;
+        }
+        if (e.isNetwork()) {
+            targetKey = MetricsKey.METRIC_REQUESTS_NETWORK_FAILED;
+        }
+        return targetKey;
```

Comment:
in the future we also cloud use  map to  store  the  relation exception to metric key

Reply:
0